limitador/src/storage/redis/redis_cached.rs

```diff
@@ -174,6 +175,9 @@ impl AsyncCounterStorage for CachedRedisStorage {
             }
         }
 
+        // ask the flusher to flush the batch
+        self.flush_tx.send(()).unwrap();
```

Comment:
I like this a lot, but would make it conditional on ... being within a certain threshold of the limit? e.g. close to expiry and/or "reaching" the limit?

Reply:
What's the benefit of delaying the flush?  I can see the following benefits: 
1.  Reduce the number Redis API calls since you group multiple calls into 1 batch
2.  Can dedupe multiple increments of the same counter

Both of the above will likely reduce CPU load on Redi s.  But that's not the main problem global rate limiters are solving for.  The main problem is high latency calls to Redis.   Adding a delay only increases that latency further.
More latency means the local cache counters will not have the correct global value.

I do think if the load is large enough CPU load on redis will be a problem, so keeping the flushing_period option is good since that would introduce that delay that can help reduce the load on Redis.  But again, I don't think that's the main problem we are solving for.